src/runtime/merge.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::{stream::Fuse, Stream, StreamExt};
+
+/// Provides the ability to merge two streams, exiting when either stream is exhausted.
+pub(crate) trait MergeExt: Stream {
+    fn merge<S>(self, other: S) -> Merge<Self, S>
+    where
+        Self: Sized,
+        S: Stream<Item = Self::Item>,
+    {
+        Merge::new(self, other)
+    }
+}
+impl<S: ?Sized> MergeExt for S where S: Stream {}
```

Comment:
Do we make use of this currently?

Reply:
Yes. This was necessary to support actors that use alternative streams. The merge from `tokio_stream` ends when ALL streams end, but we need to end when ANY stream ends.